### PR TITLE
[CIS-117] Add pre-commit hook

### PR DESCRIPTION
**Pull request by b-onc**
Please run `./bootstrap.sh` after merging this PR to install pre-commit hook!
The hook only runs on /StreamChatClient_v3 folder for now, we'll continue to use swiftlint until v3 is released.

Swiftformat rule documentation: https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md
Rules are inferred from: https://github.com/apple/swift/tree/master/stdlib/public/core

### CODE REVIEW

**Comment by b-onc** [comment]
@VojtaStavik IMHO we should do that after we release v3 and start using pre-commit hook on every commit

**Comment by VojtaStavik** [comment]
@b-onc ok with me 👍

File: .swiftformat [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Stream rules
+--header "\n{file}\nCreated on {created}.\nCopyright © {year} Stream.io Inc. All rights reserved.\n"
```
Comment (VojtaStavik) [inline .swiftformat:2]: I'd suggest removing the `Created on` from the header. It's not important when the file was created.

Comment (b-onc) [reply .swiftformat:2]: Don't you want that nostalgic feeling when looking at old files?
_Man, I remember creating this file... Has it been 4 years already?_

Comment (VojtaStavik) [reply .swiftformat:2]: no

Comment (VojtaStavik) [reply .swiftformat:2]: 😅

Comment (b-onc) [reply .swiftformat]: No good arguments from my side so I removed it

**Pull request merged**